### PR TITLE
Add `type` option

### PR DESCRIPTION
**Pull request by zbraniecki**
Solves #9

### CODE REVIEW

**Comment by zbraniecki** [comment]
@mathiasbynens and @maggiepint - that is a new development that should be included in the review.

File: spec/relativetimeformat.html [hunk 1]
```diff
@@ -100,8 +102,14 @@ <h1>PartitionRelativeTimePattern (relativeTimeFormat, value, unit)</h1>
...
-        1. Let _pr_ be ! ResolvePlural(_relativeTimeFormat_.[[PluralRules]], _value_).
-        1. Let _pattern_ be ? Get(_po_, _pr_).
+        1. Let _type be _relativeTimeFormat_.[[Type]].
+        1. If _type_ is equal to "text", then
+          1. Let _exists_ be ? HasProperty(_po_, _value_).
+          1. If _exists_ is *true*, then
+            1. Let _pattern_ be ? Get(_po_, _value_).
+        1. If _pattern_ is *undefined*, then
+          1. Let _pr_ be ! ResolvePlural(_relativeTimeFormat_.[[PluralRules]], _value_).
+          1. Let _pattern_ be ? Get(_po_, _pr_).
```
Comment (littledan) [inline spec/relativetimeformat.html]: I think rather than adding in those extra steps, we actually want to skip steps 4-5 if in numeric mode.

Comment (zbraniecki) [reply spec/relativetimeformat.html]: Which steps 4-5? When I build the spec with my patch, steps 4-5 are:
```
4. If isFinite(value) is false, then throw a RangeError exception.
5. Let unit be ? SingularRelativeTimeUnit(unit).
```

Is that what you're asking to skip?

Comment (littledan) [reply spec/relativetimeformat.html]: Sorry, I meant steps 14-15, namely

        1. Let _exists_ be ? HasProperty(_patterns_, ToString(_value_)).
        1. If _exists_ is *true*, then
          1. Let _result_ be Get(_patterns_, ToString(_value_)).
          1. Return a List containing the Record { [[Type]]: `"literal"`, [[Value]]: _result_ }.

Comment (zbraniecki) [reply spec/relativetimeformat.html]: oh, of course!

File: spec/relativetimeformat.html [hunk 2]
```diff
@@ -90,10 +92,12 @@ <h1>PartitionRelativeTimePattern (relativeTimeFormat, value, unit)</h1>
         1. If _exists_ is *false*, then
           1. Let _entry_ be _unit_.
         1. Let _patterns_ be ? Get(_fields_, _entry_).
-        1. Let _exists_ be ? HasProperty(_patterns_, ToString(_value_)).
-        1. If _exists_ is *true*, then
-          1. Let _result_ be Get(_patterns_, ToString(_value_)).
-          1. Return a List containing the Record { [[Type]]: `"literal"`, [[Value]]: _result_ }.
+        1. Let _type be _relativeTimeFormat_.[[Type]].
+        1. If _type_ is equal to "text", then
```
Comment (littledan) [inline spec/relativetimeformat.html]: Nit: Use backticks around the literal string.

**Pull request merged**